ad/movement/trusts.md

```diff
@@ -198,6 +207,50 @@ When doing NTLM authentications across trusts, the trusting domain's domain cont
+3. To enumerate the shadow security principals, its members from the current (bastion) forest and privileges in the user/production forest, one can use the following command from the ActiveDirectory module: `Get-ADObject -SearchBase ("CN=Shadow Principal Configuration,CN=Services," + (Get-ADRootDSE).configurationNamingContext) -Filter * -Properties * | select Name,member,msDS-ShadowPrincipalSid | fl`
+The following properties are the most interesting ones:
+- `name` - Name of the shadow principal
+- `member` - Members from the bastion forest which are mapped to the shadow principal. In our example, it is the Domain Administrator of defensiveps.local.
+- `msDS-ShadowPrincipalSid` - The SID of the principal (user or group) in the user/prodcution forest whose privileges are assigned to the shadow security principal.
+### Abusing the shadow principals
+1. If an attacker compromises the user listed in "member" above they can use the shadow principals. With the privileges achieved using shadow principals above, they can access the user forest using RDP (explicit credentials of the bastion user required), WMI, PowerShell Remoting etc. Please note that if Kerberos AES Encryption is not enabled for the PAM trust, an attacker needs to add the machines of the existing forest in `WSMan TrustedHosts` and use the `-Authentication Negotiate` option with PowerShell remoting cmdlets.
+2. One can also use `SIDHistor`y injection using `Mimikatz` to abuse the PAM trust.
+3. Overall, auditors should look for a way to compromise the Bastion (Red) forest to easily gain access to all the managed Production Forests.
+### Persistence
+Attackers can also use this for persistence. Please note that the persistence will be for the privileges on the user/production forest and not the Bastion Forest itself. Once the Bastion Forest is compromised, there are multiple ways to achieve persistence:
+1. One can add a user to an existing shadow security principal container. `Set-ADObject -Identity "CN=psforest-ShadowEnterpriseAdmin,CN=Shadow Principal Configuration,CN=Services,CN=Configuration,DC=defensiveps,DC=local" -Add @{'member'="CN=lowpriv user,CN=Users,DC=defensiveps,DC=local"} -Verbose`
+Please note that in this case, if someone looks at the details of the 'lowprivuser', that account would appear to be a part of the `psforest-ShadowEnterpriseAdmin`'group'.
+2. A better and more feasible TTP would be the modification of ACLs of the shadow principal object. One can provide a user in control, Full Permission overt shadow principal object but a principle of minimal permissions should always be met. `Read Members` and `Write Members` permissions on the shadow principal object are adequate to add and remove principals at will from the shadow principals. At this point, one can add or remove users at will with the privileges of 'reportdbadmin' user. On top of that, by default there are no logs for any changes to the ACL or 'membership' of a shadow principal.
```

Comment:
This needs to be moved in a separate page in the persistence section: https://www.thehacker.recipes/ad/persistence

Reply:
I have created a separate page for this TTP.